compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -555,35 +555,29 @@ object Symbols {
 
     type ThisName = TypeName
 
-    /** If this is a top-level class, and if `-Yretain-trees` is set, return the TypeDef tree
-     *  for this class, otherwise EmptyTree. This will force the info of the class.
-     */
-    def tree(implicit ctx: Context): tpd.Tree /* tpd.TypeDef | tpd.EmptyTree */ = {
+    /** If this is a top-level class, and if `-Yretain-trees` or `-Xlink-optimise` is set,
```

Comment:
This comment is not 100% true since if this is a top-level class and -Xlink-optimise is set, but the class comes from the sources and not tasty, it will return EmptyTree.